drivers/storage/portworx/component/plugin.go

```diff
@@ -0,0 +1,351 @@
+	"github.com/libopenstorage/operator/pkg/util/k8s"
+	ocpconfig "github.com/openshift/api/config/v1"
+	consolev1 "github.com/openshift/api/console/v1"
+	coreops "github.com/portworx/sched-ops/k8s/core"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	PluginComponentName       = "Portworx Plugin"
+	PluginName                = "portworx"
+	PluginDeploymentName      = "px-plugin"
+	PluginConfigMapName       = "px-plugin"
+	PluginServiceName         = "px-plugin"
+	NginxDeploymentName       = "px-plugin-proxy"
+	NginxConfigMapName        = "px-plugin-proxy-conf"
+	NginxServiceName          = "px-plugin-proxy"
+	ClusterOperatorVersion    = "config.openshift.io/v1"
+	ClusterOperatorKind       = "ClusterOperator"
+	OpenshiftAPIServer        = "openshift-apiserver"
+	OpenshiftSupportedVersion = "4.12"
+	BaseDir                   = "/configs/"
+	nginxConfigMapFileName    = "nginx-configmap.yaml"
+	nginxDeploymentFileName   = "nginx-deployment.yaml"
+	nginxServiceFileName      = "nginx-service.yaml"
+	consolePluginFileName     = "consoleplugin.yaml"
+	pluginConfigmapFileName   = "plugin-configmap.yaml"
+	pluginDeploymentFileName  = "plugin-deployment.yaml"
+	pluginServiceFileName     = "plugin-service.yaml"
+	TRUE                      = "true"
+	FALSE                     = "false"
+)
+
+type plugin struct {
+	client                    client.Client
+	scheme                    *runtime.Scheme
+	isPluginDeploymentCreated bool
+	isProxyDeploymentCreated  bool
+	isPluginSupported         string
+}
+
+func (p *plugin) Initialize(
+	k8sClient client.Client,
+	_ version.Version,
+	scheme *runtime.Scheme,
+	recorder record.EventRecorder,
+) {
+	p.client = k8sClient
+	p.scheme = scheme
+}
+
+func (c *plugin) Name() string {
+	return PluginComponentName
+}
+
+func (c *plugin) Priority() int32 {
+	return DefaultComponentPriority
+}
+
+func (c *plugin) IsPausedForMigration(cluster *corev1.StorageCluster) bool {
+	return util.ComponentsPausedForMigration(cluster)
+}
+
+func (p *plugin) IsEnabled(cluster *corev1.StorageCluster) bool {
+	if p.isPluginSupported == TRUE {
+		return true
+	} else if p.isPluginSupported == "" {
+		gvk := schema.GroupVersionKind{
+			Kind:    ClusterOperatorKind,
+			Version: ClusterOperatorVersion,
+		}
+		exists, err := coreops.Instance().ResourceExists(gvk)
+		if err != nil {
+			logrus.Error(err)
+			p.isPluginSupported = FALSE
+			return false
+		}
+
+		if exists {
+			operator := &ocpconfig.ClusterOperator{}
+			err := p.client.Get(
+				context.TODO(),
+				types.NamespacedName{
+					Name: OpenshiftAPIServer,
+				},
+				operator,
+			)
+
+			if errors.IsNotFound(err) {
```

Comment:
We are not handling all the errors here. This could lead to a nil pointer error below as the operator object could be nil on error.

Similar to above comment, let's only set the flag if we are able to successfully call the API.
```
if err != nil {
  logrus.Error(err)
  if errors.IsNotFound() {
    p.isPluginSupported = FALSE
  }
  return false
}
```